distribution/tools/entitlement-trampoline/src/main/java/org/elasticsearch/entitlement/checks/EntitlementChecks.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.checks;
+
+public interface EntitlementChecks {
+    @CheckBefore(method = "exit")
```

Comment:
For static methods, rather than having the object in the signature, I think it'll be clearer to have a class reference alongside:

`@CheckBefore(class = System.class, method = "exit")`

this could also be present for instance methods, and that would need to match a method argument type in the right place.

Reply:
Yeah, the extra argument that's always null doesn't fill me with joy either.

I opted to take the approach that simplified the instrumentation logic itself. Every "if" statement in there is another case to be tested and another opportunity for odd interactions. This approach makes everything "just work".

The extra argument is also a bit less verbose actually; and it's needed anyway for virtual methods. This approach makes them all uniform, which has some benefit.

It's also the convention used by `Field` and `Method` in the reflection API: for statics, the receiver object is null.

So, given that it was a little unclear which is better, I opted for the one that made the instrumentation code simpler.

However...

There may be cases where we want to instrument non-public classes, and we may not even be able to reference them from `EntitlementChecks`. That will require another mechanism to indicate the desired class anyway.

All in all, I don't feel super strongly about this. If folks find this revolting, I can change it.

Reply:
I would prefer that route, then we don't have to dump an extra null onto the stack for these calls.

Reply:
Based on Ryan's concern below about forcing all the classes to be loaded, I think we might want to go in a different direction here.

I think that's a pretty good reason to omit this part for now and just hard-code `System.exit` as Ryan and Lorenzo suggested.